File: LeetCode - May/singleNonDuplicate.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int l = 0 ;
        int h = nums.length -1 ;
        while (l < h)
        {
                int m = (l + h) / 2 ;

                if (m+1 <h && nums[m] != nums[m+1] && nums[m] != nums[m-1])
                    return nums[m] ; 
                
                else if( ((m & 1) == 0 && nums[m] == nums[m+1]) || ((m & 1) == 1 && nums[m] == nums[m-1]) )
                        l = m + 1; 
                else
                    h = m - 1 ; 
            }

            return nums[l] ; 
        }
    }

File: LeetCode - May/singleNonDuplicate.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        l = 0 ;
        h = len(nums) -1 ;

        while l < h:
        
            m = (l + h) // 2 ;
            if (m+1 <h and nums[m] != nums[m+1] and nums[m] != nums[m-1]):
                return nums[m]  

            elif( ((m & 1) == 0 and nums[m] == nums[m+1]) or ((m & 1) == 1 and nums[m] == nums[m-1]) ):
                l = m + 1 
            
            else:
                h = m - 1  
        
        return nums[l] 
        